cmake/CheckIncludesSymbols.cmake

```diff
@@ -53,6 +53,30 @@ if ( UINT16_T_SIZE GREATER 0 )
   set( HAVE_UINT16_T ON )
 endif ()
 
+set( CMAKE_EXTRA_INCLUDE_FILES "sys/types.h" )
```

Comment:
I'm wondering of this some comments should be added here just to make it easier for devs in the future to understand why this is here?

Reply:
This macro needs to be set up before `check_type_size()` and loses scope after execution of `check_type_size()`. This is CMake magic. I think it is clear what this part of the CMake script does. The rest is somehow CMake-technical  and I tend not to begin documenting it.